azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_externaltfs.go

```diff
@@ -0,0 +1,111 @@
+package serviceendpoint
+
+import (
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+const (
+	personalAccessTokenExternalTFS = "personal_access_token"
+)
+
+func ResourceServiceEndpointExternalTFS() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointExternalTFS, expandServiceEndpointExternalTFS)
+	r.Schema["connection_url"] = &schema.Schema{
+		Type:         schema.TypeString,
+		ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+		Required:     true,
+		Description:  "URL of the Azure DevOps organization or the TFS Project Collection to connect to.",
+	}
+	authPersonal := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			personalAccessTokenExternalTFS: {
+				Type:         schema.TypeString,
+				Required:     true,
+				DefaultFunc:  schema.EnvDefaultFunc("AZDO_PERSONAL_ACCESS_TOKEN", nil),
+				Description:  "Personal access tokens are applicable only for connections targeting Azure DevOps organization or TFS 2017 (and higher)",
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+	patHashKey, patHashSchema := tfhelper.GenerateSecreteMemoSchema(personalAccessTokenExternalTFS)
```

Comment:
hash functions has be deprecated, `patHashKey` and  `patHashSchema` can be removed.

Reply:
Code-related to the deprecated functions were removed.